Add DockerHub credentials to E2E in CI

pivotaljohn opened the pull request:
In order to avoid rate limiting errors, uses injected credentials:
- authenticate the docker client;
- authenticate image pulls for Pods deployed as part of the test (today, just the `registry2`);
- provide credentials for kubectl-buildkit.

Includes minor improvements:
- reduce output from `wget` calls to reduce that noise
- remove some dead lines
- resolves #146

jtigger committed:
Add DockerHub credentials to E2E in CI

jtigger committed:
Use kubectl supplied by minikube

jtigger committed:
Use latest version of minikube in e2e

jtigger committed:
Reduce noise in e2e output

- wget's progress indicator are useless in troubleshooting failed builds
  and take up a lot of room in the output; -nv reports success or
  failure details.
- removed false precision from CIDR for insecure registries to avoid
  misleading the reader that the IP address of the minikube VM is (or
  needs to be) static (it just needs to be [and will be] in the
  specified CIDR block).

jtigger committed:
Remove git config / download boot2docker via wget

- we're using wget to download all _other_ binaries.
- `git config` lines are suspect since no commits are being created in
  this script.

jtigger committed:
Fixed typo in new wget call

cppforlife commented on .github/workflows/test-gh.yml:
```diff
@@ -29,40 +30,53 @@ jobs:
         fetch-depth: 0
     - name: Run Tests
+      env:
+        DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
```
@pivotaljohn what account are we including creds for?

pivotaljohn replied:
Today, it's "k8slt".

pivotaljohn replied:
Updated so that the username is in plaintext. 👍🏻

jtigger committed:
Hardcode DockerHub username

jtigger committed:
Always use the latest version minikube in e2e tests

... so that if there are incompatibilities with the emerging versions of
Kubernetes, we learn about it as soon as possible.

DennisDenuto commented on .github/workflows/test-gh.yml:
```diff
@@ -29,40 +30,53 @@ jobs:
 
+        # Authenticate to DockerHub to avoid test failures due to rate limiting
+        # ... for builders using Docker
+        echo "${DOCKERHUB_ACCESS_TOKEN}" | docker login --username "${DOCKERHUB_USERNAME}" --password-stdin
```
With the above bash option`set -x` I think bash will print out the eval'd `DOCKERHUB_ACCESS_TOKEN` variable. 

for e.g.
```
bash-5.0$ FOO=test
bash-5.0$ set -x
bash-5.0$ echo $FOO | grep bar
+ echo test
+ grep bar
bash-5.0$
```

But I'm pretty sure [github actions will redact this](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-secrets). But!... If you wanted to be safer than sorry you could wrap this around `set +x / set -x`, or potentially drop the `set -x` on top? (it should make the logs less verbose, hopefully not reducing the ease of debugging by much either).

pivotaljohn replied:
Yeah, it _is_ redacted, but not interested in relying on that. I'll wrap the call as it's particularly helpful in troubleshooting to have `set -x`... and we've done _other_ work to significantly reduce the total log size. 👍🏻

DennisDenuto replied:
that makes sense! i'll resolve this convo :-)

jtigger committed:
Disable trace output while handling access token

pivotaljohn commented:
@cppforlife + @DennisDenuto : GTG?
